File: src/main/kotlin/org/smarthata/service/health/DeviceHealth.java
package org.smarthata.service.health;


import java.time.LocalDateTime;

public class DeviceHealth {

    public String devicePath;
    public DeviceStatus status;

    public LocalDateTime updateTime;
    public LocalDateTime lastNotificationTime;

    public DeviceHealth(String devicePath) {
        this.devicePath = devicePath;
        this.status = DeviceStatus.ACTIVE;
    }
}

File: src/main/kotlin/org/smarthata/service/health/DeviceHealth.kt
package org.smarthata.service.health

import java.time.LocalDateTime

data class DeviceHealth(
    var devicePath: String,
) {
    var status: DeviceStatus = DeviceStatus.UNKNOWN
    var updateTime: LocalDateTime = LocalDateTime.now()
    var lastNotificationTime: LocalDateTime? = null

    enum class DeviceStatus {
        ACTIVE, OFFLINE, UNKNOWN
    }
}
